packages/auth/src/authClients/supabase.ts

```diff
@@ -1,30 +1,37 @@
-import type {
-  SupabaseAuthUser,
-  SupabaseClient as Supabase,
-  SupabaseAuthResponse,
-} from '@supabase/supabase-js'
+import { Session, User, Provider } from '@supabase/gotrue-js/dist/main/lib/types'
+import type SupabaseClient from '@supabase/supabase-js/dist/main/SupabaseClient'
 
 import type { AuthClient } from './index'
-export type SupabaseUser = SupabaseAuthUser
-export type { Supabase }
+export type Supabase = SupabaseClient
+export type SupabaseUser = User
 
 export interface AuthClientSupabase extends AuthClient {
   login(options: {
     email: string
     password: string
```

Comment:
Should we allow for a null or optional password to support the [magic link](https://supabase.io/docs/client/auth-signin#sign-in-with-magic-link) login?

https://supabase.io/docs/client/auth-signin#sign-in-with-magic-link

Reply:
As we discussed, I wanted to get this PR in asap. I'll do a follow-up PR with magic link and oauth support.